src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -102,47 +107,35 @@ public sealed class PeerConnector : IPeerConnector
         public RelatedPeerConnectors RelatedPeerConnector { get; set; }
 
         /// <inheritdoc/>
-        public NetworkPeerRequirement Requirements { get; private set; }
+        public NetworkPeerRequirement Requirements { get; internal set; }
 
         /// <summary>Constructor used for unit testing.</summary>
-        internal PeerConnector(
-            IPeerAddressManager peerAddressManager,
-            PeerIntroductionType peerIntroductionType)
+        protected PeerConnector(NodeSettings nodeSettings, IPeerAddressManager peerAddressManager)
         {
+            Guard.NotNull(nodeSettings, nameof(nodeSettings));
             Guard.NotNull(peerAddressManager, nameof(peerAddressManager));
 
-            this.nodeLifetime = new NodeLifetime();
             this.peerAddressManager = peerAddressManager;
-            this.peerIntroductionType = peerIntroductionType;
+            this.NodeSettings = nodeSettings;
+            this.nodeLifetime = new NodeLifetime();
             this.RelatedPeerConnector = new RelatedPeerConnectors();
         }
 
-        /// <summary>Constructor used by dependency injection.</summary>
-        internal PeerConnector(Network network,
-            INodeLifetime nodeLifeTime,
-            NetworkPeerConnectionParameters parameters,
-            NetworkPeerRequirement requirements,
-            Func<IPEndPoint, byte[]> groupSelector,
-            IAsyncLoopFactory asyncLoopFactory,
-            IPeerAddressManager peerAddressManager,
-            PeerIntroductionType peerIntroductionType,
-            INetworkPeerFactory networkPeerFactory)
+        /// <summary>Constructor used by <see cref="Connection.ConnectionManager"/>.</summary>
+        protected PeerConnector(PeerConnectorContext context)
```

Comment:
If context is only used to pass types in the constructor you are only introducing complexity (both for testing and dependency injection)

Reply:
Ah I only did this for readability, as its easier to create the context once and then just pass it into the constructor. Also, PeerConnector doesn't get its constructor argument's DI-ed.

Reply:
>Also, PeerConnector doesn't get its constructor argument's

yes but maybe we will be able to find a way to do it at some point